pytorch_lightning/trainer/trainer.py

```diff
@@ -871,7 +871,11 @@ def from_argparse_args(cls, args: Union[Namespace, ArgumentParser], **kwargs) ->
         params = vars(args)
 
         # we only want to pass in valid Trainer args, the rest may be user specific
-        valid_kwargs = inspect.signature(cls.__init__).parameters
+        valid_kwargs = {}
+        trainer_cls = cls
+        while trainer_cls != Trainer:
```

Comment:
what would be the case you want Trainer subclass?

Reply:
I haven't personally encountered a need to subclass `Trainer`, but I thought some other might have (I don't know about any reason why we shouldn't, if there are).

My main motivation to apply it to both `LightningDataModule` and `Trainer` was to keep the behavior the same. The methods with the same name behave the same way, and I thought it should be kept that way. I saw [talks in this issue](https://github.com/PyTorchLightning/pytorch-lightning/issues/2707) that some might want to refactor the argparse methods to avoid duplicating them. So I figured having similar but not identical functions would be more trouble in the end.